python/pyspark/sql/connect/client/core.py

```diff
@@ -220,7 +220,9 @@ def userId(self) -> Optional[str]:
 
     @property
     def token(self) -> Optional[str]:
-        return self._params.get(ChannelBuilder.PARAM_TOKEN, None)
+        return self._params.get(
+            ChannelBuilder.PARAM_TOKEN, os.environ.get("SPARK_CONNECT_AUTHENTICATE_TOKEN")
```

Comment:
We should probably don't set this if `os.environ.get("SPARK_CONNECT_AUTHENTICATE_TOKEN")` is `None` because we will not use this when this auth is off.